src/clients/KubectlClient.ts

```diff
@@ -100,62 +100,48 @@ export class KubectlClient implements IClient {
         }
     }
 
-    public async getContainersList(podName: string): Promise<string[]> {
-        if (!podName) {
-            this._logger.error(TelemetryEvent.KubectlClient_GetPodNameError, new Error(`Pod name is null`));
+    public async getContainersList(podNameList: string[], namespace: string): Promise<string[]> {
```

Comment:
I think this should take a single pod, so that this code doesn't need to make decisions about how the containers for multiple pods get combined (which may vary depending on callers).

See #5